### PR TITLE
More fixes

### PR DESCRIPTION
**Pull request by akallabeth**
* Adds some more assertions/checks
* Adds basic multitransport message parsing

### CODE REVIEW

File: libfreerdp/core/rdp.c [hunk 1]
```diff
@@ -1971,7 +1986,7 @@ rdpRdp* rdp_new(rdpContext* context)
 	if (!rdp->heartbeat)
 		goto fail;
 
-	rdp->multitransport = multitransport_new();
+	rdp->multitransport = multitransport_new(rdp, INITIATE_REQUEST_PROTOCOL_UDPFECL);
```
Comment (hardening) [inline libfreerdp/core/rdp.c:1990]: Lossy is the old version not used in first intention by the servers, I'd suggest `INITIATE_REQUEST_PROTOCOL_UDPFECR | INITIATE_REQUEST_PROTOCOL_UDPFECR`

File: libfreerdp/core/multitransport.c [hunk 2]
```diff
@@ -17,35 +17,155 @@
...
+{
+	rdpRdp* rdp;
+
+	UINT32 requestId;         /* requestId (4 bytes) */
+	UINT16 requestedProtocol; /* requestedProtocol (2 bytes) */
+	UINT16 reserved;          /* reserved (2 bytes) */
+	BYTE securityCookie[16];  /* securityCookie (16 bytes) */
+};
+
+static BOOL compare(const rdpMultitransport* srv, const rdpMultitransport* client)
```
Comment (hardening) [inline libfreerdp/core/multitransport.c:39]: Perhaps we can have a less generic name than `compare` ...

Comment (akallabeth) [reply libfreerdp/core/multitransport.c:39]: `true`, did not think about that as the function is `static`

File: libfreerdp/core/rdp.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/core/rdp.c:1990]: @hardening typo? you meant `INITIATE_REQUEST_PROTOCOL_UDPFECL | INITIATE_REQUEST_PROTOCOL_UDPFECR` right?

**Comment by hardening** [comment]
Can we wait before merging this one, I'd like to influence the multi-transport parsing so that it matches my own changes with UDP and multi-transport...

File: libfreerdp/core/multitransport.c [hunk 3]
```diff
@@ -17,35 +17,155 @@
...
 #define TAG FREERDP_TAG("core.multitransport")
 
-int rdp_recv_multitransport_packet(rdpRdp* rdp, wStream* s)
+struct rdp_multitransport
+{
+	rdpRdp* rdp;
+
+	UINT32 requestId;         /* requestId (4 bytes) */
+	UINT16 requestedProtocol; /* requestedProtocol (2 bytes) */
+	UINT16 reserved;          /* reserved (2 bytes) */
```
Comment (hardening) [inline libfreerdp/core/multitransport.c:35]: We can avoid storing reserved that is not used at all.

**Comment by akallabeth** [comment]
@hardening ok, then we merge this first? I´ll mark it ready then.

File: libfreerdp/core/multitransport.c (hunk 3 above)
Comment (akallabeth) [reply libfreerdp/core/multitransport.c:35]: true, was just for completeness

**Review by MartinHaimberger — APPROVED** [review]
LGTM

**Pull request merged**